src/coreclr/jit/emitxarch.cpp

```diff
@@ -5646,6 +5646,19 @@ void emitter::emitStoreSimd12ToLclOffset(unsigned varNum, unsigned offset, regNu
         // Store upper 4 bytes
         emitIns_S_R(INS_movss, EA_4BYTE, tmpReg, varNum, offset + 8);
     }
+    else
+    {
+        // We don't have temp regs - let's do two shuffles then
+
+        // [0,1,2,3] -> [2,3,0,1]
+        emitIns_R_R_I(INS_pshufd, EA_16BYTE, dataReg, dataReg, 78);
+
+        //  Store upper 4 bytes
+        emitIns_S_R(INS_movss, EA_4BYTE, dataReg, varNum, offset + 8);
+
+        // Restore dataReg to its previous state: [2,3,0,1] -> [0,1,2,3]
+        emitIns_R_R_I(INS_pshufd, EA_16BYTE, dataReg, dataReg, 78);
+    }
```

Comment:
I don't think so, in this case I couldn't find a place where I could reserve a temp reg for this since the tree is created too late during LSRA itself - since it's not too bad compared to the "with tmp reg" solution and is only taken when SSE41 is not presented (and the diffs are only on 32bits) I decided to leave as is.

Reply:
Actually no, it is needed for arm64 as well